chore: creating initial upgrade file (#471)
terraform-google-modules/terraform-example-foundation: docs/upgrading_to_v2.0.md

# Upgrade Guidance

Before moving forward with adopting components of V2, please review the list of
breaking changes below. You can find a list of all changes in the
[Changelog](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/CHANGELOG.md).

**Note:** There is no in-place upgrade path from v1 to v2.

## Breaking Changes

-  The repo now requires Terraform version 0.13 (minimum). For v1 the minimum version was
   0.12.x.
-  V2 introduces a new alternative hub-and-spoke network architecture,

>    described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf). 

Reviewer: ```suggestion
   described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf).
```

> -  In V2, the infrastructure pipeline has transitioned from using 

Reviewer: ```suggestion
-  In V2, the infrastructure pipeline has transitioned from using
```

# Upgrade Guidance

Before moving forward with adopting components of V2, please review the list of
breaking changes below. You can find a list of all changes in the
[Changelog](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/CHANGELOG.md).

> Note: There is no in-place upgrade path from v1 to v2.

Reviewer: ```suggestion
**Note:** There is no in-place upgrade path from v1 to v2.
```

## Breaking Changes

-  The repo now requires Terraform version 0.13 (minimum). For v1 the minimum version was
   0.12.x.
-  V2 introduces a new alternative hub-and-spoke network architecture,

>    described in Section 7.2 of the blueprint. The Terraform to build the

Reviewer: ```suggestion
   described in Section 7.2 of the [Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf). The Terraform to build the
```

>    described in Section 7.2 of the blueprint. The Terraform to build the

Reviewer: what does `The Terraform` referes to ? 
are they scripts or code?

Reviewer 2: removed

   hub-and-spoke network are located in 1-org and 3-networks.

> -  In V2, the infrastructure pipeline has transitioned from using Google
>    Container Registry (GCR) to using Google Artifact Registry. Artifact

Reviewer: ```suggestion
-  In V2, the infrastructure pipeline has transitioned from using 
[Google Container Registry](https://cloud.google.com/container-registry/docs) (GCR) to using [Google Artifact Registry](https://cloud.google.com/artifact-registry/docs). Artifact
```

   Registry extends the capabilities of GCR as outlined
   [here](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr#compare).

>    The Terraform can be found in 4-projects, within the infra_pipeline module.

Reviewer: what does `The Terraform` referes to ?

Reviewer 2: I assume it means the configs are located in those stages, but I think we can remove this.

> -  [Some Regular Firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical FW rules](https://cloud.google.com/vpc/docs/firewall-policies), which provides the same functionality of allowing or denying connections to or from your virtual machine instances but allowing enforcement of consistent firewall policies across your organization.

Author: [Some VPC firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical firewall policy rules]

> -  [Some Regular Firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical FW rules](https://cloud.google.com/vpc/docs/firewall-policies), which provides the same functionality of allowing or denying connections to or from your virtual machine instances but allowing enforcement of consistent firewall policies across your organization.

Reviewer: ```suggestion
-  Some [VPC firewall rules](https://cloud.google.com/vpc/docs/firewalls) have been replaced with [Hierarchical firewall policy rules](https://cloud.google.com/vpc/docs/firewall-policies), which provides the same functionality of allowing or denying connections to or from your virtual machine instances but allowing enforcement of consistent firewall policies across your organization.
```

as per @mark1000 suggestion

## Steps to upgrade codebase

> Note: When you run `terraform apply`, expect resources to be deleted and recreated. Be sure

Reviewer: ```suggestion
**Note:** When you run `terraform apply`, expect resources to be deleted and recreated. Be sure
```
